sys/include/net/gnrc/netif.h

```diff
@@ -246,15 +247,10 @@ struct gnrc_netif_ops {
  * @note If @ref DEVELHELP is defined netif_params_t::name is used as the
  *       name of the network interface's thread.
  *
- * @attention   Fails and crashes (assertion error with @ref DEVELHELP or
- *              segmentation fault without) if `GNRC_NETIF_NUMOF` is lower than
- *              the number of calls to this function.
- *
  * @return  The network interface on success.
  */
-gnrc_netif_t *gnrc_netif_create(char *stack, int stacksize, char priority,
-                                const char *name, netdev_t *dev,
-                                const gnrc_netif_ops_t *ops);
+gnrc_netif_t *gnrc_netif_create(gnrc_netif_t *netif, char *stack, int stacksize,
```

Comment:
This type of return value no longer makes sense. It is simply the `netif` pointer as specified as the `netif` parameter.

The return value should be an `int`, which is 0 if successful, or any negative value if an error occurs. The error code could be that returned by `thread_create`.

On the other hand, changing the return value here would require changing the return value of all `gnrc_netif_*_create' functions.

Reply:
It totally makes sense. Since I already changed the API of these functions, I don't think it will be a problem to change it again

Reply:
done!